src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/AssemblyIdentityMustMatchTests.cs

```diff
@@ -30,170 +36,106 @@ public static void AssemblyNamesDoNotMatch()
+            IAssemblySymbol rightSymbol = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
 
             // Compatible assembly versions
             ApiComparer differ = new();
-            IEnumerable<CompatDifference> differences = differ.GetDifferences(leftSymbols, rightSymbols);
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(leftSymbol, rightSymbol);
             Assert.Empty(differences);
 
-            ApiComparer strictDiffer = new();
-            strictDiffer.StrictMode = true;
-            leftSymbols = new AssemblySymbolLoader().LoadAssembly(leftDllPath);
-            rightSymbols = new AssemblySymbolLoader().LoadAssembly(rightDllPath);
+            differ.StrictMode = true;
 
             // Not strictly compatible
-            differences = strictDiffer.GetDifferences(leftSymbols, rightSymbols);
+            differences = differ.GetDifferences(leftSymbol, rightSymbol);
             Assert.Single(differences);
 
-            CompatDifference expected = new CompatDifference(DiagnosticIds.AssemblyIdentityMustMatch, string.Empty, DifferenceType.Changed, "Project, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null");
+            CompatDifference expected = new CompatDifference(DiagnosticIds.AssemblyIdentityMustMatch, string.Empty, DifferenceType.Changed, $"{leftSymbol.Name}, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null");
             Assert.Equal(expected, differences.First(), CompatDifferenceComparer.Default);
         }
 
-        [RequiresMSBuildVersionTheory("17.0.0.32901")]
+        [Theory]
         [InlineData(true)]
         [InlineData(false)]
         public void AssemblyKeyTokenMustBeCompatible(bool strictMode)
         {
-            var testAsset = _testAssetsManager
-                .CopyTestAsset("PublicKeyTokenValidation", allowCopyIfPresent: true)
-                .WithSource();
+            string syntax = "namespace EmptyNs { }";
 
-            BuildCommand buildCommand = new(testAsset);
-            buildCommand.Execute().Should().Pass();
-
-            string leftDllPath = Path.Combine(buildCommand.GetOutputDirectory("netstandard2.0").FullName, "PublicKeyTokenValidation.dll");
-            string rightDllPath = Path.Combine(buildCommand.GetOutputDirectory("net6.0").FullName, "PublicKeyTokenValidation.dll");
-            IAssemblySymbol leftSymbols = new AssemblySymbolLoader().LoadAssembly(leftDllPath);
-            IAssemblySymbol rightSymbols = new AssemblySymbolLoader().LoadAssembly(rightDllPath);
+            IAssemblySymbol leftSymbol = SymbolFactory.GetAssemblyFromSyntax(syntax, publicKey: _publicKey);
+            IAssemblySymbol rightSymbol = SymbolFactory.GetAssemblyFromSyntax(syntax, publicKey: _publicKey);
 
             // public key tokens must match
             ApiComparer differ = new();
             differ.StrictMode = strictMode;
 
-            IEnumerable<CompatDifference> differences = differ.GetDifferences(leftSymbols, rightSymbols);
```

Comment:
can we add a check here that leftAssemblyKeyToken is not null

Reply:
Yup. I'll add a check in all tests for that kind of information.